src/Contract/Operation/Reductionable.php

```diff
@@ -18,11 +19,16 @@
 interface Reductionable
 {
     /**
-     * Reduce a collection of items through a given callback.
+     * Reduce a collection of items through a given callback
+     * and yield each intermediary results.
      *
-     * @param mixed $initial
+     * @template V
+     * @template W
      *
-     * @return Collection<TKey, T>
+     * @param callable(V|W, T, TKey, Iterator<TKey, T>): W $callback
+     * @param V $initial
+     *
+     * @return Collection<TKey, V|W>
```

Comment:
shouldn't these be the same as `Reduceable`? Instead here we have 2 parameter types

https://github.com/loophp/collection/blob/master/src/Contract/Operation/Reduceable.php#L24-L29

Reply:
Indeed, you're right.

Reply:
Fixed in b44b87f689da7e7f9a0ac205302a4b697347bb22